### PR TITLE
push down aggregations to gitbase

### LINKED ISSUE #6: Pushdown rules
Rules for community version:
- [x] join pushdown
- [x] #20 aggregations pushdown
- [x] #19 functions pushdown (UAST functions LANGUAGE function)

### LINKED ISSUE #20: Aggregation pushdown
Related to #6 

This task has two parts:
- [x] Tree pushdown
- [x] Aggregation pushdown

### PR DESCRIPTION
**Pull request by erizocosmico**
Depends on #26 
Closes #6 
Closes #20 

Note that tests fail because one of the tests in DefaultSourceSpec fails due to "unable to connect to bblfsh server", but that was already failing before.

### CODE REVIEW

**Comment by erizocosmico** [comment]
Tests will pass once https://github.com/src-d/gitbase-spark-connector/pull/29 is merged

File: src/main/scala/tech/sourced/gitbase/spark/Query.scala [hunk 1]
```diff
@@ -18,17 +18,67 @@ import org.apache.spark.sql.catalyst.expressions.{
...
+      subquery,
+      source,
+      removeSourcesFromSort(sort),
+      removeSources(grouping)
+    )
+  }
 
   /**
     * Creates a new query replacing the projected columns with the given ones.
+    * If there is already a projection, it will be treated as a subquery.
```
Comment (mcarmonaa) [inline src/main/scala/tech/sourced/gitbase/spark/Query.scala]: I guess the previous line doesn't apply anymore if columns aren't really replaced. Why is necessary keep the old projections in a subquery?

**Comment by ajnavarro** [comment]
@erizocosmico #29 already merged.

**Comment by erizocosmico** [comment]
Yep, waiting on #26 to not have to rebase twice

**Comment by ajnavarro** [comment]
Superseeded by: #34

**Pull request closed without merging**